### PR TITLE
Add support for ClassPrivateProperty

### PR DESCRIPTION
**Pull request by existentialism**
Fixes https://github.com/prettier/prettier/issues/2810.

(I'll add `ClassPrivateMethod` in a separate PR since it's a diff Babylon plugin, and diff stage)

### CODE REVIEW

**Comment by suchipi** [comment]
Looks like there are a few files with formatting violations, based on the travis results- could you reformat those with prettier 1.7.0? Once travis is passing, I think we are good to merge

File: tests/classes_private_fields/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  #y;
+}
+
+class Point {
+  #x = 1;
+  #y = 2;
+
+  constructor(x = 0, y = 0) {
+    this.#x = +x;
+    this.#y = +y;
```
Comment (vjeux) [inline tests/classes_private_fields/__snapshots__/jsfmt.spec.js.snap]: unrelated but the syntax looks really weird.

Comment (suchipi) [reply tests/classes_private_fields/__snapshots__/jsfmt.spec.js.snap]: Yeah... this proposal has a lot of history behind it 😓 https://github.com/tc39/proposal-private-fields/issues/14

**Comment by azz** [comment]
This is by far my least favourite proposal that has advanced through tc39. :(

File: tests/classes_private_fields/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (bakkot) [reply tests/classes_private_fields/__snapshots__/jsfmt.spec.js.snap]: That issue isn't the half of it - I have some meeting notes from *1999* exploring how we might add private fields. Turns out it's hard; it's a miracle we figured out a way of doing it at all in a language without static types.

**Pull request merged**